### PR TITLE
fix job active count;  count terminating pod as failed

### PR DESCRIPTION
**Pull request by yowenter**
if pod is terminating , the pod.Status.Phase is still RUNNING. 
In this case, the Job status maybe not correct.

### CODE REVIEW

File: pkg/core/status.go [hunk 1]
```diff
@@ -34,7 +34,14 @@ func InitializeReplicaStatuses(jobStatus *apiv1.JobStatus, rtype apiv1.ReplicaTy
...
 	case corev1.PodRunning:
-		jobStatus.ReplicaStatuses[rtype].Active++
+		if pod.DeletionTimestamp != nil {
+			// when node is not ready, the pod will be in terminating state.
+			// Count deleted Pods as failures to account for orphan Pods that
+			// never have a chance to reach the Failed phase.
+			jobStatus.ReplicaStatuses[rtype].Failed++
+		} else {
+			jobStatus.ReplicaStatuses[rtype].Active++
+		}
```
Comment (tenzen-y) [inline pkg/core/status.go:44]: I don't think that the deleting pod is failed one. Please let me know if I'm missing any cases.

Comment (yowenter) [reply pkg/core/status.go:44]: Hi, @tenzen-y, 

I found some code in `pkg/util/k8sutil/k8sutil.go` 
```
func IsPodActive(p *v1.Pod) bool {
	return v1.PodSucceeded != p.Status.Phase &&
		v1.PodFailed != p.Status.Phase &&
		p.DeletionTimestamp == nil
}
```

, it seems that job active count should not contains the pod in `Terminating` state.
And in kubernetes jobs implementation, the `Terminating` pod doesn't count as active either.
In some cases, if the node becomes not ready, we are not certain if  the pod is RUNNING or not...

Comment (tenzen-y) [reply pkg/core/status.go:44]: Uhm. You have a point. When the pod is running and has a DeletionTimestamp field, we should mark Failed to the Job.

Thank you for clarifying!

**Comment by tenzen-y** [comment]
@yowenter And also, can you update the unit test?

https://github.com/kubeflow/common/blob/master/pkg/controller.v1/common/status_test.go

**Comment by yowenter** [comment]
> @yowenter And also, can you update the unit test?
> 
> https://github.com/kubeflow/common/blob/master/pkg/controller.v1/common/status_test.go

Hi, @tenzen-y,  I've added job status unittest, please review again.

File: pkg/controller.v1/common/status_test.go [hunk 2]
```diff
@@ -2,10 +2,13 @@ package common
 
 import (
 	"testing"
+	"time"
 
 	apiv1 "github.com/kubeflow/common/pkg/apis/common/v1"
 	"github.com/stretchr/testify/assert"
 	corev1 "k8s.io/api/core/v1"
+
```
Comment (tenzen-y) [inline pkg/controller.v1/common/status_test.go]: nit: revert this line.

File: pkg/controller.v1/common/status_test.go [hunk 3]
```diff
@@ -14,13 +17,14 @@ func TestUpdateJobReplicaStatuses(t *testing.T) {
 	_, ok := jobStatus.ReplicaStatuses["worker"]
 	// assert ReplicaStatus for "worker" exists
 	assert.True(t, ok)
-	setStatusForTest(&jobStatus, "worker", 2, 3, 1)
-	assert.Equal(t, jobStatus.ReplicaStatuses["worker"].Failed, int32(2))
+	setStatusForTest(&jobStatus, "worker", 2, 3, 1, 1)
+	// terminating pod should count as failed.
```
Comment (tenzen-y) [inline pkg/controller.v1/common/status_test.go:20]: good comment :)

**Review by tenzen-y — COMMENTED** [review]
@yowenter Looks great! Thank you!

/lgtm
/assign @terrytangyuan

**Comment by tenzen-y** [comment]
/lgtm cancel 
for nit

**Review by terrytangyuan — APPROVED** [review]
/approve

**Pull request merged**